src/main/resources/static/js/front.js

```diff
@@ -9,45 +9,99 @@ function cleanForm() {
 
 function cleanResults() {
     $("#resultLink").html("").attr("href", "");
-    $("#result").addClass('invisible');
+    $("#result").addClass("invisible");
+
+    $("#qrCode img").attr("src", "");
+    $("#qrCode").addClass("invisible");
 }
 
 function showError(errorMessage) {
     $("#errorText").html(errorMessage);
-    $("#errorModal").modal('show');
+    $("#errorModal").modal("show");
 }
 
-function sendLink(long_url) {
-    var storeLinkRoute = "/api/store";
+function calculateQRCodeSize() {
+    var browserWidth = $(window).width();
+    var defaultQRBlockSize = 371;
+    var defaultQRCodeSize = 350;
+    var qrBlockRatio = 0.943; //350/371
 
-    var longUrl = long_url.trim();
-    var json = {
-        link: longUrl
-    };
-    doPost(storeLinkRoute, json, onSuccessStoreLink, onFailStoreLink);
+    var size;
+    if (browserWidth > defaultQRBlockSize) {
+        size = defaultQRCodeSize;
+    } else {
+        size = browserWidth * qrBlockRatio;
+    }
+
+    return size;
 }
 
+function onFailGenerateQRCode(jqXHR, textStatus, errorThrown) {
+    showError("Internal error. Got malformed reply from QR generator");
+    $("#qrCode img").attr("src", "");
+    $("#qrCode").addClass("invisible");
+
+    if (jqXHR !== null) {
+        var replyRaw = jqXHR.responseText;
+        console.debug("QR Code Reply JSON: " + replyRaw);
+        console.debug("QR Code Reply TextStatus: " + textStatus);
+        console.debug("QR Code Reply ErrorThrown: " + errorThrown);
+    }
+}
+
+function onSuccessGenerateQRCode(data, textStatus, jqXHR) {
+    if (jqXHR.status === 200) {
+        var qrCode = data.qrCode;
+        if (qrCode.trim().length === 0) {
+            showError("Internal error. Got malformed reply from QR generator");
+            return;
+        }
+
+        $("#qrCode img").attr("src", qrCode);
+        $("#qrCode").removeClass("invisible");
+    }
+}
+
+function updateCounter() {
+    var counter = $("#overallLinksNum");
+    var currentNum = counter.text();
+
+    if ($.isNumeric(currentNum)) {
+        counter.text(parseInt(currentNum, 10) + 1);
+    } else {
+        console.error("Failed to update counter. Current counter value is not a number");
+    }
+}
+
+function generateQRCode(ident) {
+    var size = calculateQRCodeSize();
+    var qrCodeGeneratorRoute = "/api/qrCode/" + ident + "/" + parseInt(size, 10);
+    doGet(qrCodeGeneratorRoute, onSuccessGenerateQRCode, onFailGenerateQRCode);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['doGet' is not defined.](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=5487058)